packages/twenty-front/src/modules/object-record/record-board/record-board-card/components/RecordBoardCellFieldInput.tsx

```diff
@@ -0,0 +1,73 @@
+import { useRecordBoardColumnContextOrThrow } from '@/object-record/record-board/record-board-column/contexts/RecordBoardColumnContext';
+import { FieldInput } from '@/object-record/record-field/components/FieldInput';
+import { FieldContext } from '@/object-record/record-field/contexts/FieldContext';
+import { useIsFieldValueReadOnly } from '@/object-record/record-field/hooks/useIsFieldValueReadOnly';
+import { FieldInputEvent } from '@/object-record/record-field/types/FieldInputEvent';
+import { getDropdownFocusIdForRecordField } from '@/object-record/utils/getDropdownFocusIdForRecordField';
+import { getRecordFieldInputId } from '@/object-record/utils/getRecordFieldInputId';
+import { activeDropdownFocusIdState } from '@/ui/layout/dropdown/states/activeDropdownFocusIdState';
+import { useContext } from 'react';
+import { useRecoilCallback } from 'recoil';
+
+export const RecordBoardCellFieldInput = () => {
+  const { recordId, fieldDefinition } = useContext(FieldContext);
+  const { onUpsertRecord, onCloseInlineCell } =
+    useRecordBoardColumnContextOrThrow();
+  const isFieldReadOnly = useIsFieldValueReadOnly();
+
+  const handleEnter: FieldInputEvent = (persistField) => {
+    persistField();
+    onUpsertRecord({
```

Comment:
shouldn't we call useUpsertBoardRecord here?

Looks like the main challenge:
- if we wouldn't have the opporunity (+ company case), we only upsert the record once the name is filled (so calling  useUpsertBoardRecord here), I think that's what's done on the RecordTable
- but as we need to persist the company too, you are persisting the record sooner and only updating it here now.

I would try to call useUpsertBoardRecord (so no persist it sooner), and instead for the opportunity.company case, I would write in the recordStore { recordId: pendingRecordId, company: xxx, position: xxx } in useCreateNewBoardRecord

It should work!